README.md

```diff
@@ -1,61 +1,2577 @@
+quicker to write". But beware! Whilst the second form of `CMD` does indeed
+require you to use a JSON structure, it *also* ensures that signals
+passed to the container are propagated to it. That means that when the
+container is stopped using `docker stop`, the `SIGTERM` signal is correctly
+passed to the script or executable that allows it to clean up correctly. Using
+the 'shell' form will cause the process to be effectively be `SIGKILL`ed.
+
+As such, you should always use the 'exec' form unless there are very good
+reasons for not doing so.
+
+As the `CMD` directive informs Docker of what to run when a container is
+started, there should only ever be *one* `CMD` directive in a Dockerfile. If
+more than one exist, only the *last* `CMD` directive seen will be run. Sometimes
+this can be useful, if you want to use a base image which usually runs as a
+service, but wish to override the `CMD` directive from that base image (perhaps
+you want to run with different parameters, or use the base image because of the
+filesystem layers it includes).
+
+The `CMD` directive can also be used in conjunction with the `ENTRYPOINT`
+directive, which we'll describe in a more advanced Docker Masterclass.
+
+### 5. Building a Docker Image
+
+Now that we know some of the basic Dockerfile directives, it's worth learning
+how to put some of these together to build a simple Docker image.
+
+For convenience, export a variable to point to the root of this masterclass
+repository, as we'll use this for the rest of the exercises, e.g.:
+```
+$ export BALENA_DOCKER_MASTERCLASS=~/docker-masterclass
+```
+Now change directory to the simple-docker-app directory in the root of this
+masterclass repository, e.g.:
+```
+$ cd $BALENA_DOCKER_MASTERCLASS/simple-docker-app
+```
+Note that for the rest of these exercises, we're assuming that you're using an
+x64 based machine. Should you be using an alternative architecture, such as
+Arm, you'll need to change the architecture of the base image that you're using
+in the Dockerfiles.
+
+We're going to use `balenalib` based images in the remainder of the exercises in
+this Masterclass, as they include useful images which are 'ready to go' with
+various languages, libraries, etc.
+
+Load a code editor of your choice, and edit the `Dockerfile` file in the
+`simple-docker-app` directory. Add the following:
+```
+FROM balenalib/intel-nuc-debian-node:latest
+
+COPY * /usr/src/app/
+
+RUN cd /usr/src/app \
+    && npm install
+
+CMD ["/usr/local/bin/node", "/usr/src/app/index.js"]
+```
+Ensure you have a newline at the end of the Dockerfile. Let's quickly run
+through exactly what this Dockerfile does:
+
+`FROM balenalib/intel-nuc-debian-node:latest`
+This declares that the Docker image built should inherit from the
+`balenalib/intel-nuc-debian-node:latest` image, which in this case is the latest
+Debian Node.js image built and tagged by balena. This image includes all
```

Comment:
If you don't explicitly tag an image, then it automatically becomes the `latest`. If you do explicitly tag the image `latest`, then you're saying 'this is the latest good version of the image we've built`, even if this is an older version, because you may end up having to roll back an image if for some reason a later version regresses. The key words here a 'latest good', because latest absolutely doesn't imply the time it was built. It's just the 'latest' good version of a release. As such, I'm keeping this as is unless there's a better argument against it. :)

Reply:
I understand r.e. the Docker side and I was more coming from an English angle of the word "latest" which implies most recent in time. I'm reminded of articles like this https://vsupalov.com/docker-latest-tag/. That being said, keep it like it is 😆